maven/lib/dependabot/maven/new_version.rb

```diff
@@ -0,0 +1,68 @@
+# typed: strict
```

Comment:
Did we consider integrating with Maven core? This logic seems like a duplication of a logic that Maven itself must be implementing somewhere

Reply:
No we did not consider integrating with Maven core. My gut feeling is that this is less complex than an integration.

Reply:
Perhaps less complex, but it'll be a long-term maintenance and syncing effort

Ie: Maven 4 is coming soon, and this specification may change

Reply:
@abdulapopoola The short answer is that I do not know. My point is that this will be a long term maintenance overhead when Maven parses this internally. There must be a way to integrate maven SDK here and avoid the duplication. 

Given that this was merged I suppose that it can be a follow up

Reply:
This may help in the future : https://maven.apache.org/pom.html#version-order-testing